### ISSUE TITLE
BLIP client blocks indefinitely when peer abruptly closes connection

### ISSUE THREAD

**Issue by tleyden** [couchbase/go-blip#17]
In https://github.com/couchbase/sync_gateway/issues/3268, it was observed that in certain cases, Sync Gateway will abruptly close the websocket connection. which is considered a separate issue being tracked in https://github.com/couchbase/sync_gateway/issues/3221

The high level is that the test code connecting to the Sync Gateway blip replicator is seen to block indefinitely in the case where Sync Gateway abruptly closes the connection.

I've added a test to attempt to repro which isolates this to pure go-blip code.  Here is the test:

https://github.com/couchbase/go-blip/blob/510b853303b4e955d77c1a2a9848161c1aca2e17/context_test.go#L14-L109

Does this look like a bug?  Or is this expected behavior?

The real concern is that if the conversation was flipped, and Sync Gateway was sending an outbound request to a Couchbase Mobile client which abruptly terminated a connection, would Sync Gateway's handler goroutine end up permanently blocked on the call to `request.Response()`?  If so, that would be a severe resource leak that would need to get addressed.

**Comment by snej** [comment]
Your last paragraph makes sense; this seems like something that should be fixed.

I'm assuming that the `receiver` object does get notified that the WebSocket closed. So there probably needs to be some mechanism added by which it can wake up those blocked Response() calls, which would then return a synthesized error response.
